ada_feeding_perception/ada_feeding_perception/segment_from_point.py

```diff
@@ -191,6 +212,28 @@ def initialize_food_segmentation(self) -> None:
         )
         self.get_logger().info("...Done!")
 
+    def camera_info_callback(self, msg: CameraInfo) -> None:
+        """
+        Store the latest camera info message.
+
+        Parameters
+        ----------
+        msg: The camera info message.
+        """
+        with self.camera_info_lock:
+            self.camera_info = msg
+
+    def depth_image_callback(self, msg: Image) -> None:
+        """
+        Store the latest depth image message.
+
+        Parameters
+        ----------
+        msg: The depth image message.
+        """
+        with self.latest_depth_img_msg_lock:
+            self.latest_depth_img_msg = msg
+
     def image_callback(self, msg: Image) -> None:
```

Comment:
As above, not needed for this PR, but flagging as an Issue. We should make this `msg: Union[CompressedImage, Image]` to support either arg type.

That can apply to `segment_image` as well, using `image_msg.__class__ == CompressedImage` as an if statement to swap between the two bridge functions.

And for the subscriber, you can glean which topic type (and also check the topic type at runtime) by calling `self.get_topic_names_and_types()`, [searching through the list of tuples for the topic name](https://stackoverflow.com/questions/2917372/how-to-search-a-list-of-tuples-in-python), and checking the type name for `sensor_msgs/CompressedImage` or `sensor_msgs/Image`. Perhaps breaking it off into a helper function.

Reply:
Added it in, yet to test.